### PR TITLE
braintree-web: fix `number` field is optional (Fix #56167)

### LINKED ISSUE #56167: [braintree-web] CVV only case : number optional
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/51e462b9789948003fd7f9ba73663885ef5b381c/types/braintree-web/modules/hosted-fields.d.ts#L46

As per [docs](https://developer.paypal.com/braintree/docs/guides/hosted-fields/faq#can-i-verify-the-cvv-when-a-customer-uses-or-updates-a-vaulted-card?) there is a case for which number is undefined. 

Could that be changed to this pls? 
Let me know if you want me to PR

```
    number?: HostedFieldsField | undefined;
```

### PR DESCRIPTION
**Pull request by rhysd**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [link](https://developer.paypal.com/braintree/docs/guides/hosted-fields/faq#can-i-verify-the-cvv-when-a-customer-uses-or-updates-a-vaulted-card?)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Fixes #56167

### CODE REVIEW

File: types/braintree-web/modules/hosted-fields.d.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export interface HostedFieldFieldOptions {
     expirationDate?: HostedFieldsField | undefined;
     expirationMonth?: HostedFieldsField | undefined;
     expirationYear?: HostedFieldsField | undefined;
-    number: HostedFieldsField;
+    number?: HostedFieldsField;
```
Comment (martin-badin) [inline types/braintree-web/modules/hosted-fields.d.ts:46]: Please update this

```suggestion
    number?: HostedFieldsField | undefined;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/braintree-web/modules/hosted-fields.d.ts (hunk 1 above)
Comment (rhysd) [reply types/braintree-web/modules/hosted-fields.d.ts]: Done: abaead92af29b1b4f0009347a857000e61e9c4a9

**Review by martin-badin — APPROVED** [review]
Looks good

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Pull request merged**